examples/vision/semantic_segmentation_deeplab_v3.py

```diff
@@ -0,0 +1,340 @@
+"""
+Title: Semantic segmentation using DeepLabV3+
+Author: [Alexia Audevart](https://github.com/aaudevart)
+Date created: 2023/13/04
+Last modified: 2023/13/04
+Description: Use the DeepLabV3+ pre-trained models via the Keras CV API for Semantic Segmentation.
+Accelerator: GPU
+"""
+
+"""
+## Introduction
+
+Semantic segmentation, with the goal to assign semantic labels to every pixel in an
+image, is an essential computer vision task.
+
+In this example, we implement
+the **DeepLabV3+** model for semantic segmentation in healthcare. This model is a
+fully-convolutional
+architecture that performs well on semantic segmentation benchmarks.
+
+The goal is to detect blood vessels on 2D PAS-stained histology images from healthy human
+kidney tissue slides.
+
+
+### References:
+
+- [Encoder-Decoder with Atrous Separable Convolution for Semantic Image
+Segmentation](https://arxiv.org/abs/1802.02611)
+- [Rethinking Atrous Convolution for Semantic Image
+Segmentation](https://arxiv.org/abs/1706.05587)
+- [DeepLab: Semantic Image Segmentation with Deep Convolutional Nets, Atrous Convolution,
+and Fully Connected CRFs](https://arxiv.org/abs/1606.00915)
+
+### Dataset:
+
+We will use a simplified extract of the [HuBMAP - Hacking the Human Vasculature
+Dataset](https://www.kaggle.com/competitions/hubmap-hacking-the-human-vasculature)
+for training our model. This dataset is obtained from Periodic acid-Schiff (PAS) stained
+tissue slides from human kidneys.
+Each image is labeled with pixel-wise annotations for 2 categories: background and blood
+vessel.
+This dataset focuses on microvasculature in the kidney, including arterioles, capillaries
+and venules. Larger vessels structures, i.e. arteries and veins, were not segmented in
+this dataset.
+"""
+
+"""shell
+!pip install keras-core --upgrade
+!pip uninstall -y keras-cv
+!pip install git+https://github.com/keras-team/keras-cv.git
+
+!pip install gdown
+
+# This sample uses Keras Core, the multi-backend version of Keras.
+# The selected backend is TensorFlow (other supported backends are 'jax' and 'torch')
+import os
+os.environ['KERAS_BACKEND'] = 'tensorflow'
+"""
+
+import os
+import cv2
+import numpy as np
+from glob import glob
+import matplotlib.pyplot as plt
+import tensorflow as tf
+
+
+import keras_core as keras
+
+import keras_cv
+
+from keras_cv.models import ResNet18V2Backbone, DeepLabV3Plus
+
+print("TensorFlow version:", tf.__version__)
+print("KerasCV version:", keras_cv.__version__)
+
+"""
+## Accelerator
+
+Detect hardware, return appropriate distribution strategy
+"""
+
+try:
```

Comment:
For the example, I think we can exclude this bit and forego a strategy altogether. TF will just default to using the GPU in this case so we don't need to worry about distribution